app/controllers/course/assessment/question/text_responses_controller.rb

```diff
@@ -52,13 +63,30 @@ def destroy
   def text_response_question_params
     params.require(:question_text_response).permit(
       :title, :description, :staff_only_comments, :maximum_grade, :allow_attachment,
-      :hide_text,
+      :hide_text, :is_comprehension,
       skill_ids: [],
-      solutions_attributes: [:_destroy, :id, :solution_type, :solution, :grade, :explanation]
+      solutions_attributes: [:_destroy, :id, :solution_type, :solution, :grade, :explanation],
+      groups_attributes:
+      [
+        :_destroy, :id, :maximum_group_grade,
+        points_attributes:
+        [
+          :_destroy, :id, :point_grade,
+          solutions_attributes:
+          [
+            :_destroy, :id, :solution_type, :explanation, solution: []
+          ]
+        ]
+      ]
     )
   end
 
   def question_type
     @text_response_question.question_type
   end
+
+  def build_at_least_one_group_one_point
```

Comment:
Let's shift this into the model with `after_initialize, :build_at_least_one_group, if: :new_record?`. And another for the group model if there are no points.

Reply:
Okay. Thanks for the suggestion!

Reply:
This code will remain: https://github.com/Coursemology/coursemology2/pull/2755#discussion_r166821569